exonum/src/messages/helpers.rs

```diff
@@ -0,0 +1,87 @@
+use encoding::{Error, Field};
+use storage::StorageValue;
+use super::{Message, RawTransaction, SignedMessage};
+use serde::{de, Serialize, Serializer, Deserialize, Deserializer};
+
+///Transaction binary form, can be converted
+pub trait BinaryForm: Sized {
+    /// Converts transaction into serialized form.
+    fn serialize(self) -> Result<Vec<u8>, Error>;
+
+    /// Converts serialized byte array into transaction.
+    fn deserialize(buffer: &[u8]) -> Result<Self, Error>;
+}
+
+impl<T> BinaryForm for T
+where
+    T: StorageValue + for<'a> Field<'a>,
+{
+    fn serialize(self) -> Result<Vec<u8>, Error> {
+        Ok(self.into_bytes())
+    }
+
+    fn deserialize(buffer: &[u8]) -> Result<Self, Error> {
+        <Self as Field>::check(
+            buffer,
+            0.into(),
+            <Self as Field>::field_size().into(),
+            <Self as Field>::field_size().into(),
+        ).map(|_| StorageValue::from_bytes(buffer.into()))
+    }
+}
+
+
+/// Serializes `Message<RawTranasction>` as hex value.
+pub(crate) struct HexTransaction;
+impl HexTransaction {
+    pub(crate) fn serialize<S>(message: &Message<RawTransaction>, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        let hex_string = message.to_hex_string();
+        hex_string.serialize(serializer)
+    }
+
+    pub(crate) fn deserialize<'a, D>(deserializer: D) -> Result<Message<RawTransaction>, D::Error>
```

Comment:
Why did you choose not to move these methods into serde traits implementation? They are `pub(crate)` anyway

Reply:
Because it's a wrapper between our encoding_struct and serde, later this trait should be removed.